packages/ajna-contracts/contracts/ajna/ajna-actions/AjnaProxyActions.sol

```diff
@@ -328,10 +328,16 @@ contract AjnaProxyActions is IAjnaProxyActions {
         _pull(debtToken, amount);
         IERC20(debtToken).safeApprove(address(pool), amount);
         (, , , , , uint256 lupIndex_) = poolInfoUtils.poolPricesInfo(address(pool));
+        uint256 balanceBefore = IERC20(debtToken).balanceOf(address(this));
         pool.repayDebt(address(this), amount * pool.quoteTokenScale(), 0, address(this), lupIndex_);
         if (stamploan) {
             pool.stampLoan();
         }
+        uint256 repaidAmount = balanceBefore - IERC20(debtToken).balanceOf(address(this));
```

Comment:
So there is a prospect problem here:
- Using the balance of the `debtToken` in the contract can be dangerous. ERC777, which is backwards compatible with ERC20, allow the owner of the tokens to be called back when a transfer happens. The owner could then send more debt tokens to the pool contract, modifying the actual balance. I don't think it can be used to drain tokens, but it would definitely make the call fail if the substraction operation is less than 0 (implicit underflow check in Solidity) for line 336
- Also the `leftOverBalance` variable is unsigned, so it cannot be negative, rendering the `if` in line 338 useless

I've checked and the `repayDebt` call from Ajna does not return anything (unfortunately). I think the best approach is to take the balance of debt token BEFORE pulling the tokens in (base balance) and then return anything that exceeds that balance. This preserves the invariant: "`repayDebt` preserves the number of `debtTokens` in the contract"

Reply:
Discussed offline and decided that it is fine. If more tokens are sent to the contract the transaction will revert which is OK, and the if for greater than zero is to prevent zero transfers of the tokens